RecoTracker/LSTCore/src/alpaka/Hit.h

```diff
@@ -0,0 +1,267 @@
+      hitRangesUpper = alpaka::getPtrNative(hitsbuf.hitRangesUpper_buf);
+      hitRangesnLower = alpaka::getPtrNative(hitsbuf.hitRangesnLower_buf);
+      hitRangesnUpper = alpaka::getPtrNative(hitsbuf.hitRangesnUpper_buf);
+    }
+  };
+
+  template <typename TDev>
+  struct hitsBuffer : hits {
+    Buf<TDev, unsigned int> nHits_buf;
+    Buf<TDev, float> xs_buf;
+    Buf<TDev, float> ys_buf;
+    Buf<TDev, float> zs_buf;
+    Buf<TDev, uint16_t> moduleIndices_buf;
+    Buf<TDev, unsigned int> idxs_buf;
+    Buf<TDev, unsigned int> detid_buf;
+    Buf<TDev, float> rts_buf;
+    Buf<TDev, float> phis_buf;
+    Buf<TDev, float> etas_buf;
+    Buf<TDev, float> highEdgeXs_buf;
+    Buf<TDev, float> highEdgeYs_buf;
+    Buf<TDev, float> lowEdgeXs_buf;
+    Buf<TDev, float> lowEdgeYs_buf;
+    Buf<TDev, int> hitRanges_buf;
+    Buf<TDev, int> hitRangesLower_buf;
+    Buf<TDev, int> hitRangesUpper_buf;
+    Buf<TDev, int8_t> hitRangesnLower_buf;
+    Buf<TDev, int8_t> hitRangesnUpper_buf;
+
+    template <typename TQueue, typename TDevAcc>
+    hitsBuffer(unsigned int nModules, unsigned int nMaxHits, TDevAcc const& devAccIn, TQueue& queue)
+        : nHits_buf(allocBufWrapper<unsigned int>(devAccIn, 1u, queue)),
+          xs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          ys_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          zs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          moduleIndices_buf(allocBufWrapper<uint16_t>(devAccIn, nMaxHits, queue)),
+          idxs_buf(allocBufWrapper<unsigned int>(devAccIn, nMaxHits, queue)),
+          detid_buf(allocBufWrapper<unsigned int>(devAccIn, nMaxHits, queue)),
+          rts_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          phis_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          etas_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          highEdgeXs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          highEdgeYs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          lowEdgeXs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          lowEdgeYs_buf(allocBufWrapper<float>(devAccIn, nMaxHits, queue)),
+          hitRanges_buf(allocBufWrapper<int>(devAccIn, nModules * 2, queue)),
+          hitRangesLower_buf(allocBufWrapper<int>(devAccIn, nModules, queue)),
+          hitRangesUpper_buf(allocBufWrapper<int>(devAccIn, nModules, queue)),
+          hitRangesnLower_buf(allocBufWrapper<int8_t>(devAccIn, nModules, queue)),
+          hitRangesnUpper_buf(allocBufWrapper<int8_t>(devAccIn, nModules, queue)) {
+      alpaka::memset(queue, hitRanges_buf, 0xff);
+      alpaka::memset(queue, hitRangesLower_buf, 0xff);
+      alpaka::memset(queue, hitRangesUpper_buf, 0xff);
+      alpaka::memset(queue, hitRangesnLower_buf, 0xff);
+      alpaka::memset(queue, hitRangesnUpper_buf, 0xff);
+      alpaka::wait(queue);
+    }
+  };
+
+  // Alpaka does not support log10 natively right now.
```

Comment:
I'd suggest to move all general-purpose functions to e.g. `HeterogeneousCore/AlpakaInterface`.